cocotb/share/lib/vpi/VpiCbHdl.cpp

```diff
@@ -367,26 +368,47 @@ int VpiSignalObjHdl::set_signal_value_str(std::string &value)
     value_s.value.str = &writable[0];
     value_s.format = vpiStringVal;
 
-    return set_signal_value(value_s);
+    return set_signal_value(value_s, action);
 }
 
-int VpiSignalObjHdl::set_signal_value(s_vpi_value value_s)
+int VpiSignalObjHdl::set_signal_value(s_vpi_value value_s, gpi_set_action_t action)
 {
     FENTER
-
+    PLI_INT32 vpi_put_flag = -1;
     s_vpi_time vpi_time_s;
 
     vpi_time_s.type = vpiSimTime;
     vpi_time_s.high = 0;
     vpi_time_s.low  = 0;
 
-    if (vpiStringVar == vpi_get(vpiType, GpiObjHdl::get_handle<vpiHandle>())) {
-        // assigning to a vpiStringVar only seems to work with vpiNoDelay
+    switch (action) {
+    case GPI_DEPOSIT:
+      if (vpiStringVar == vpi_get(vpiType, GpiObjHdl::get_handle<vpiHandle>())) {
```

Comment:
Nit: this block should be indented 2 more spaces, our indents are 4 and not 2 spaces